projects/ion/src/lib/tooltip/tooltip.directive.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  ApplicationRef,
+  ComponentFactoryResolver,
+  ComponentRef,
+  Directive,
+  ElementRef,
+  EmbeddedViewRef,
+  HostListener,
+  Injector,
+  Input,
+  OnDestroy,
+} from '@angular/core';
+import {
+  TooltipColorScheme,
+  TooltipPosition,
+  TooltipTrigger,
+} from '../core/types';
+import { SafeAny } from '../utils/safe-any';
+import { TooltipComponent } from './tooltip.component';
+import { getPositions } from './utilsTooltip';
+
+@Directive({
+  selector: '[ionTooltip]',
+})
+export class TooltipDirective implements OnDestroy {
+  @Input() ionTooltipTitle = '';
+  @Input() ionTooltipColorScheme: TooltipColorScheme = 'dark';
+  @Input() ionTooltipPosition: TooltipPosition = TooltipPosition.DEFAULT;
+  @Input() ionTooltipArrowPointAtCenter = true;
+  @Input() ionTooltipTrigger: TooltipTrigger = TooltipTrigger.DEFAULT;
+  @Input() ionTooltipShowDelay = 0;
+
+  private componentRef: ComponentRef<TooltipComponent> = null;
+  private delayTimeout: number;
+
+  constructor(
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private appRef: ApplicationRef,
+    private injector: Injector,
+    private elementRef: ElementRef
+  ) {}
+
+  createComponent(): HTMLElement {
+    const componentFactory =
+      this.componentFactoryResolver.resolveComponentFactory(TooltipComponent);
+
+    this.componentRef = componentFactory.create(this.injector);
+
+    this.appRef.attachView(this.componentRef.hostView);
+
+    return (this.componentRef.hostView as EmbeddedViewRef<SafeAny>)
+      .rootNodes[0];
+  }
+
+  setComponentProperties(): void {
+    if (this.componentRef !== null) {
+      this.componentRef.instance.ionTooltipTitle = this.ionTooltipTitle;
+      this.componentRef.instance.ionTooltipColorScheme =
+        this.ionTooltipColorScheme;
+      this.componentRef.instance.ionTooltipPosition = this.ionTooltipPosition;
+
+      this.delayTimeout = window.setTimeout(
+        this.showTooltip.bind(this),
+        this.ionTooltipShowDelay
+      );
+
+      this.setComponentPosition();
+    }
+  }
+
+  setComponentPosition(): void {
+    const { left, right, top, bottom } =
+      this.elementRef.nativeElement.getBoundingClientRect();
+
+    const hostPositions = { left, right, top, bottom };
+    const positions = getPositions(
+      hostPositions,
+      this.ionTooltipArrowPointAtCenter
+    );
+
+    this.componentRef.instance.left = positions[this.ionTooltipPosition].left;
+    this.componentRef.instance.top = positions[this.ionTooltipPosition].top;
+  }
+
+  attachComponentToView(): void {
+    document.body.appendChild(this.createComponent());
+    this.setComponentProperties();
+  }
+
+  showTooltip(): void {
+    if (this.componentRef !== null) {
+      this.componentRef.instance.ionTooltipVisible = true;
+    }
+  }
+
+  destroyComponent(): void {
+    if (this.componentRef !== null) {
```

Comment:
What do you think of create a function to return this?

`this.componentRef === null`

This will help when a change happen in this validation and will help in read, I think.. What do you think?

Reply:
Sure, it's a good idea!